app/lib/submission_builder/return_header.rb

```diff
@@ -12,10 +16,10 @@ def document
           xml.TaxPeriodEndDt date_type(Date.new(@submission.data_source.tax_return_year, 12, 31))
         end
         xml.TaxYr @submission.data_source.tax_return_year
-        if @submission.data_source.state_code.upcase == "NJ"
+        if !state_submission_builder.ptin.nil? && !state_submission_builder.preparer_person_name.nil?
```

Comment:
[dust] - would this work the same and avoid a double-negative?

`if state_submission_builder.ptin.present? && state_submission_builder.preparer_person_name.present?`